File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun inorderTraversal(root: TreeNode?): List<Int> {
        val result = mutableListOf<Int>()
        inorder(root, result)
        return result
    }
    
    fun inorder(root: TreeNode?, result: MutableList<Int>){
        if(root != null){
            inorder(root.left, result)
            result.add(root.`val`)
            inorder(root.right, result)
        }
    }
}

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
    var array = [];
    inOrder(root, array);
    return array;
};

function inOrder(root, array) {
    if(root != null){
        inOrder(root.left, array);
        array.push(root.val);
        inOrder(root.right, array);
    }
}